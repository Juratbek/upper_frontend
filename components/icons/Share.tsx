import { IIconProps } from './types';

export const ShareIcon = ({ width = 24, height = 24, color = 'black' }: IIconProps) => (
  <svg width={width} height={height} viewBox='0 0 24 24' fill='none'>
    <path
      d='M10.5959 13.3974L22.5997 2.78917M10.5959 13.3974L2.55763 8.95484C2.40182 8.86873 2.4277 8.63727 2.59868 8.5877L22.5997 2.78917M10.5959 13.3974L13.8261 21.7852C13.8902 21.9516 14.1235 21.9572 14.1956 21.7942L22.5997 2.78917'
      stroke={color}
      strokeWidth='1.6'
    />
  </svg>
);
