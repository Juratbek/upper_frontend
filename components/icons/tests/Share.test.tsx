import { render, screen } from '@testing-library/react';

import { ShareIcon } from '../Share';

describe('Share icon', () => {
  it('renders the icon', () => {
    render(<ShareIcon />);
    const icon = screen.getByRole('icon');
    expect(icon).toMatchSnapshot();
  });
});
