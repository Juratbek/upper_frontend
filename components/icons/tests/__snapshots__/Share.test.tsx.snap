// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Share icon > renders the icon 1`] = `
<svg
  fill="none"
  height="24"
  role="icon"
  viewBox="0 0 24 24"
  width="24"
>
  <path
    d="M10.5959 13.3974L22.5997 2.78917M10.5959 13.3974L2.55763 8.95484C2.40182 8.86873 2.4277 8.63727 2.59868 8.5877L22.5997 2.78917M10.5959 13.3974L13.8261 21.7852C13.8902 21.9516 14.1235 21.9572 14.1956 21.7942L22.5997 2.78917"
    stroke="black"
    stroke-width="1.6"
  />
</svg>
`;
